pkg/descheduler/framework/plugins/loadaware/utilization_util.go

```diff
@@ -243,40 +320,97 @@ func evictPodsFromSourceNodes(
 	continueEviction continueEvictionCond,
 	evictionReasonGenerator evictionReasonGeneratorFn,
 ) {
-	var targetNodes []*corev1.Node
-	totalAvailableUsages := map[corev1.ResourceName]*resource.Quantity{}
-	for _, destinationNode := range destinationNodes {
-		targetNodes = append(targetNodes, destinationNode.node)
-
-		for _, resourceName := range resourceNames {
-			quantity, ok := totalAvailableUsages[resourceName]
-			if !ok {
-				switch resourceName {
-				case corev1.ResourceCPU:
-					quantity = resource.NewMilliQuantity(0, resource.DecimalSI)
-				case corev1.ResourceMemory, corev1.ResourceEphemeralStorage, corev1.ResourceStorage:
-					quantity = resource.NewQuantity(0, resource.BinarySI)
-				default:
-					quantity = resource.NewQuantity(0, resource.DecimalSI)
-				}
-				totalAvailableUsages[resourceName] = quantity
+	totalAvailableUsages, targetNodes := targetAvailableUsage(destinationNodes, resourceNames, false)
+	prodAvailableUsages, prodTargetNodes := targetAvailableUsage(prodDestinationNodes, resourceNames, true)
+	bothTotalAvailableUsage, bothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, false)
+	prodBothAvailableUsage, prodBothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, true)
+
+	nodeTotalAvailableUsages := newAvailableUsage(resourceNames)
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			if _, totalOk := totalAvailableUsages[resourceName]; totalOk {
+				quantity.Add(*totalAvailableUsages[resourceName])
+			}
+			if _, prodOk := bothTotalAvailableUsage[resourceName]; prodOk {
+				quantity.Add(*bothTotalAvailableUsage[resourceName])
 			}
-			quantity.Add(*destinationNode.thresholds.highResourceThreshold[resourceName])
-			quantity.Sub(*destinationNode.usage[resourceName])
 		}
 	}
+	nodeKeysAndValues := []interface{}{
+		"nodePool", nodePoolName,
+	}
+	for resourceName, quantity := range nodeTotalAvailableUsages {
+		nodeKeysAndValues = append(nodeKeysAndValues, string(resourceName), quantity.String())
+	}
+	klog.V(4).InfoS("Total node usage capacity to be moved", nodeKeysAndValues...)
+
+	targetNodes = append(targetNodes, bothTotalNodes...)
+	balancePods(ctx, nodePoolName, sourceNodes, targetNodes, nodeTotalAvailableUsages, dryRun, nodeFit, false, resourceWeights, podEvictor,
+		podFilter, nodeIndexer, continueEviction, evictionReasonGenerator)
 
-	keysAndValues := []interface{}{
+	prodTotalAvailableUsages := newAvailableUsage(resourceNames)
+	for _, resourceName := range resourceNames {
+		if quantity, ok := prodTotalAvailableUsages[resourceName]; ok {
+			if _, totalOk := prodAvailableUsages[resourceName]; totalOk {
+				quantity.Add(*prodAvailableUsages[resourceName])
+			}
+			if _, prodOk := prodBothAvailableUsage[resourceName]; prodOk {
```

Comment:
`prodBothAvailableUsage` rootly was calculated from the same nodes where `bothTotalAvailableUsage` was calculated, that is, `bothDestinationNodes`. If we migrate pods from `targetNodes` to `bothDestinationNodes` in the first `balancePods`, some `bothDestinationNodes` might changed and cannot be as a Destination, even if there is still space in `prodBothAvailableUsage`.

Reply:
need to re-calculate `prodTotalAvailableUsages`

Reply:
The corresponding calculation method has been updated.